### PR TITLE
refactor: EventSenders forward RequestToJoin decision to control node

### PR DESCRIPTION
**Pull request by 0x-r4bbit**
This is a bigger change in how community membership requests are handled among admins, token masters, owners, and control nodes.

Prior to this commit, all privileged users, also known as `EventSenders`, were able to accept and reject community membership requests and those changes would be applied by all users.

This commit changes this behaviour such that:

1. EventSenders can make a decision (accept, reject), but merely forward their decision to the control node, which ultimately has to confirm it
2. EventSenders are no longer removing or adding members to and from communities
3. When an eventsender signaled a decision, the membership request will enter a pending state (acceptedPending or rejectedPending)
4. Once a decision was made by one eventsender, no other eventsender can override that decision

This implementation is covered with a bunch of tests:

- Ensure that decision made by event sender is shared with other event senders
  - `testAcceptMemberRequestToJoinResponseSharedWithOtherEventSenders()`
  - `testRejectMemberRequestToJoinResponseSharedWithOtherEventSenders()`
- Ensure memebrship request stays pending, until control node has confirmed decision by event senders
  - `testAcceptMemberRequestToJoinNotConfirmedByControlNode()`
  - `testRejectMemberRequestToJoinNotConfirmedByControlNode()`
- Ensure that decision made by event sender cannot be overriden by other event senders
  - `testEventSenderCannotOverrideRequestToJoinState()`

These test cases live in three test suites for different event sender types respectively

- `OwnerWithoutCommunityKeyCommunityEventsSuite`
- `TokenMasterCommunityEventsSuite`
- `AdminCommunityEventsSuite`

In addition to the changes mentioned above, there's also a smaller changes that ensures membership requests to *not* attached revealed wallet addresses when the requests are sent to event senders (in addition to control nodes).

Requests send to a control node will still include revealed addresses as the control node needs them to verify token permissions.

This commit does not yet handle the case of event senders attempting to kick and ban members.

Similar to accepting and rejecting membership requests, kicking and banning need a new pending state. However, we don't track such state in local databases yet so those two cases will be handled in future commit to not have this commit grow larger.

### CODE REVIEW

File: protocol/activity_center.go [hunk 1]
```diff
@@ -37,6 +37,8 @@ const (
 	ActivityCenterMembershipStatusPending
 	ActivityCenterMembershipStatusAccepted
 	ActivityCenterMembershipStatusDeclined
+	ActivityCenterMembershipStatusAcceptedPending
+	ActivityCenterMembershipStatusDeclinedPending
```
Comment (0x-r4bbit) [inline protocol/activity_center.go:41]: @status-im/desktop-devs these are new states that desktop UI should be aware of

File: protocol/communities/community.go [hunk 2]
```diff
@@ -824,9 +824,8 @@ func (o *Community) RemoveUserFromOrg(pk *ecdsa.PublicKey) (*protobuf.CommunityD
 		}
 	}
 
-	o.removeMemberFromOrg(pk)
-
 	if isControlNode {
+		o.removeMemberFromOrg(pk)
```
Comment (0x-r4bbit) [inline protocol/communities/community.go]: This is moved into the condition because admins should not be able to remove users like that, We'll need a pending state for this.

For now, we at least prevent the member to be removed by the admin

File: protocol/communities/community.go [hunk 3]
```diff
@@ -1995,31 +1994,14 @@ func (o *Community) AddMemberWithRevealedAccounts(dbRequest *RequestToJoin, role
 	defer o.mutex.Unlock()
 
 	isControlNode := o.IsControlNode()
-	allowedToSendEvents := o.HasPermissionToSendCommunityEvents()
 
-	if !isControlNode && !allowedToSendEvents {
+	if !isControlNode {
```
Comment (0x-r4bbit) [inline protocol/communities/community.go:1943]: Admins should not make changes to members directly, so only control node is allowed to do stuff here.
Also, we want to move away from attaching revealed addresses to members in the community description (as this is public metadata).

Next step is to remove usage of this function entirely.

File: protocol/communities/community_event.go [hunk 4]
```diff
@@ -160,7 +160,6 @@ func (o *Community) ToCommunityRequestToJoinAcceptCommunityEvent(changes *Commun
 	return &CommunityEvent{
 		CommunityEventClock:    o.NewCommunityEventClock(),
 		Type:                   protobuf.CommunityEvent_COMMUNITY_REQUEST_TO_JOIN_ACCEPT,
-		MembersAdded:           changes.MembersAdded,
```
Comment (0x-r4bbit) [inline protocol/communities/community_event.go:163]: No need attach `MembersAdded` if admins can't actually do that anymore. 
Now we're just signalling which request to join is marked as accepted

File: protocol/communities/community_event.go [hunk 5]
```diff
@@ -315,26 +314,6 @@ func (o *Community) updateCommunityDescriptionByCommunityEvent(communityEvent Co
...
-		}
-
-	case protobuf.CommunityEvent_COMMUNITY_REQUEST_TO_JOIN_REJECT:
-		for pkString := range communityEvent.RejectedRequestsToJoin {
-			pk, err := common.HexToPubkey(pkString)
-			if err != nil {
-				return err
-			}
-			o.removeMemberFromOrg(pk)
-		}
```
Comment (0x-r4bbit) [inline protocol/communities/community_event.go:344]: Removal/addition of members via admin events is now done later in `handleAdditionalAdminChanges()` where the control node makes the ultimate decision

File: protocol/communities/manager.go [hunk 6]
```diff
@@ -1370,65 +1371,144 @@ func (m *Manager) HandleCommunityEventsMessage(signer *ecdsa.PublicKey, message
...
+			// We only want to do this when `requestUpdated == true`, if the control node
+			// didn't know this request before, it means it just learned about it via this
+			// community event, which in turn means, we don't have any revealed addresses
+			// for this request yet, so we don't actually want to perform an
+			// accept action by the control node
+			acceptedRequestsToJoin = append(acceptedRequestsToJoin, requestToJoin.ID)
+		}
+	}
+	if community.IsControlNode() {
+		m.publish(&Subscription{AcceptedRequestsToJoin: acceptedRequestsToJoin})
```
Comment (0x-r4bbit) [inline protocol/communities/manager.go:1459]: This event causes `Messenger` to `AcceptCommunityRequestToJoin()` for all accepted requests

File: protocol/communities/manager.go [hunk 7]
```diff
@@ -1601,54 +1685,85 @@ func (m *Manager) AcceptRequestToJoin(request *requests.AcceptRequestToJoinCommu
...
+			return community, ErrNoPermissionToJoin
+		}
+
+		memberRoles := []protobuf.CommunityMember_Roles{}
+		if isAdmin {
+			memberRoles = []protobuf.CommunityMember_Roles{protobuf.CommunityMember_ROLE_ADMIN}
+		}
+
+		// TODO: Don't add member to community, instead store revealed accounts in database
+		_, err = community.AddMemberWithRevealedAccounts(dbRequest, memberRoles, revealedAccounts)
```
Comment (0x-r4bbit) [inline protocol/communities/manager.go:1730]: This needs to be removed as part of https://github.com/status-im/status-desktop/issues/11573

File: protocol/communities/request_to_join.go [hunk 8]
```diff
@@ -16,6 +16,8 @@ const (
 	RequestToJoinStateDeclined
 	RequestToJoinStateAccepted
 	RequestToJoinStateCanceled
+	RequestToJoinStateAcceptedPending
+	RequestToJoinStateDeclinedPending
```
Comment (0x-r4bbit) [inline protocol/communities/request_to_join.go:20]: @status-im/desktop-devs also new states we probably want to use in the UI

File: protocol/messenger_communities.go [hunk 9]
```diff
@@ -284,6 +284,18 @@ func (m *Messenger) handleCommunitiesSubscription(c chan *communities.Subscripti
...
+					for _, requestID := range sub.AcceptedRequestsToJoin {
+						accept := &requests.AcceptRequestToJoinCommunity{
+							ID: requestID,
+						}
+						_, err := m.AcceptRequestToJoinCommunity(accept)
+						if err != nil {
+							m.logger.Warn("failed to accept request to join ", zap.Error(err))
+						}
+					}
+				}
```
Comment (0x-r4bbit) [inline protocol/messenger_communities.go:282]: This will be triggered as a side effect in `HandleCommunityEventsMessage` when there were accepted requests to join

File: protocol/activity_center.go (hunk 1 above)
Comment (osmaczko) [reply protocol/activity_center.go:41]: I am wondering why we need this notification. Isn't pending state shared only among admins and control node? If that's the case, I believe we don't need a notification for that :thinking:

File: protocol/communities/manager.go [hunk 10]
```diff
@@ -1370,65 +1371,144 @@ func (m *Manager) HandleCommunityEventsMessage(signer *ecdsa.PublicKey, message
...
+			acceptedRequestsToJoin = append(acceptedRequestsToJoin, requestToJoin.ID)
+		}
+	}
+	if community.IsControlNode() {
+		m.publish(&Subscription{AcceptedRequestsToJoin: acceptedRequestsToJoin})
+	}
+	return nil
+}
+
+func (m *Manager) handleCommunityEventRequestRejected(community *Community, communityEvent *CommunityEvent) error {
```
Comment (mprakhov) [inline protocol/communities/manager.go:1464]: Does the control node need to notify privileged users about the declined request, as you do for an accepted one?

File: protocol/activity_center.go (hunk 1 above)
Comment (MishkaRogachev) [reply protocol/activity_center.go:41]: How we should show those states in the UI?

Comment (0x-r4bbit) [reply protocol/activity_center.go:41]: Everything related to UI is discussed here: https://github.com/status-im/status-desktop/issues/11649

File: protocol/communities_events_owner_without_community_key_test.go [hunk 11]
```diff
@@ -40,6 +41,10 @@ func (s *OwnerWithoutCommunityKeyCommunityEventsSuite) GetEventSender() *Messeng
 	return s.ownerWithoutCommunityKey
 }
 
+func (s *OwnerWithoutCommunityKeyCommunityEventsSuite) GetEventSender2() *Messenger {
+	return s.ownerWithoutCommunityKey2
+}
+
```
Comment (mprakhov) [inline protocol/communities_events_owner_without_community_key_test.go]: I wonder if it is possible to create this user in the common test scenario. The same as we did for the `User` in requests to join tests

Comment (mprakhov) [reply protocol/communities_events_owner_without_community_key_test.go]: Like here
https://github.com/status-im/status-go/pull/3843/files#diff-2990ef21a6e7432ed6cdc95cc05b9f47439c44742fb1da050adb30d976ea1aa7R840-R845

**Review by mprakhov — COMMENTED** [review]
Great work.
Left few comments

File: protocol/communities/manager.go (hunk 10 above)
Comment (jrainville) [reply protocol/communities/manager.go:1464]: They should notify the privileged users yes, so that they can delete the request and stop being in the pending state.
We still shouldn't send it to the user requesting however.

File: protocol/communities/manager.go [hunk 12]
```diff
@@ -1370,65 +1371,144 @@ func (m *Manager) HandleCommunityEventsMessage(signer *ecdsa.PublicKey, message
...
+		}
+
+		if requestUpdated {
+			// We only want to do this when `requestUpdated == true`, if the control node
+			// didn't know this request before, it means it just learned about it via this
+			// community event, which in turn means, we don't have any revealed addresses
+			// for this request yet, so we don't actually want to perform an
+			// accept action by the control node
+			acceptedRequestsToJoin = append(acceptedRequestsToJoin, requestToJoin.ID)
+		}
```
Comment (osmaczko) [inline protocol/communities/manager.go]: What if we receive many (same) requests to join through community events?

File: protocol/communities/manager.go [hunk 13]
```diff
@@ -1370,65 +1371,144 @@ func (m *Manager) HandleCommunityEventsMessage(signer *ecdsa.PublicKey, message
...
+
+	existingRequestToJoin, err := m.persistence.GetRequestToJoin(requestToJoin.ID)
+	if err != nil && err != sql.ErrNoRows {
+		return updated, err
+	}
+
+	if existingRequestToJoin != nil {
+		// node already knows about this request to join, so let's compare clocks
+		// and update it if necessary
+		if existingRequestToJoin.Clock <= requestToJoin.Clock {
```
Comment (osmaczko) [inline protocol/communities/manager.go:1395]:
```suggestion
		if existingRequestToJoin.Clock < requestToJoin.Clock {
```

I know it was like that before, but I believe same clock should be ignored.

File: protocol/communities/manager.go [hunk 14]
```diff
@@ -1601,54 +1685,85 @@ func (m *Manager) AcceptRequestToJoin(request *requests.AcceptRequestToJoinCommu
...
+		if !permissionsSatisfied {
+			return community, ErrNoPermissionToJoin
+		}
+
+		memberRoles := []protobuf.CommunityMember_Roles{}
+		if isAdmin {
+			memberRoles = []protobuf.CommunityMember_Roles{protobuf.CommunityMember_ROLE_ADMIN}
+		}
+
+		// TODO: Don't add member to community, instead store revealed accounts in database
```
Comment (osmaczko) [inline protocol/communities/manager.go]: Not sure I got it right. When/where do we add members to community then?

File: protocol/communities/manager.go [hunk 15]
```diff
@@ -1822,22 +1959,36 @@ func (m *Manager) HandleCommunityRequestToJoin(signer *ecdsa.PublicKey, request
...
+			requestToJoin.State = RequestToJoinStateAccepted
+		} else {
+			err = m.markRequestToJoinAsAcceptedPending(signer, community)
+			if err != nil {
+				return nil, err
+			}
+			requestToJoin.State = RequestToJoinStateAcceptedPending
 		}
-		// Don't check permissions here,
-		// it will be done further in the processing pipeline.
```
Comment (osmaczko) [inline protocol/communities/manager.go:1804]: This comment could still be useful.

File: protocol/communities/manager.go (hunk 10 above)
Comment (0x-r4bbit) [reply protocol/communities/manager.go:1464]: That's a good point. I also just realized we're not updating other admins to update their pending state on accepted either.

File: protocol/communities_events_owner_without_community_key_test.go (hunk 11 above)
Comment (0x-r4bbit) [reply protocol/communities_events_owner_without_community_key_test.go]: I will take a look at this!

**Review by osmaczko — COMMENTED** [review]
Nice work, although I have few questions/uncertainties.

File: protocol/communities/manager.go (hunk 12 above)
Comment (0x-r4bbit) [reply protocol/communities/manager.go]: The first one wins. But this can indeed cause a state where 2 admins look at different pending states until the ultimate state was confirmed by the control node.

To avoid that, we'd have to also store which account made which decision

File: protocol/communities/manager.go (hunk 14 above)
Comment (0x-r4bbit) [reply protocol/communities/manager.go]: They need to be store in a separated database table instead of the community descriptoin so they aren't shared with public metadata.

Then, revealed addresses only locally known need to be backed up on waku such that control nodes can recover them.

File: protocol/communities/manager.go (hunk 15 above)
Comment (0x-r4bbit) [reply protocol/communities/manager.go:1804]: Okay will undo this

File: protocol/communities/manager.go (hunk 14 above)
Comment (osmaczko) [reply protocol/communities/manager.go]: > They need to be store in a separated database table instead of the community descriptoin so they aren't shared with public metadata.

How will clients then construct the member list?

Comment (0x-r4bbit) [reply protocol/communities/manager.go]: Ah sorry, the comment has a mistake. This is not about the members but about the revealed addresses

File: protocol/communities/manager.go (hunk 12 above)
Comment (osmaczko) [reply protocol/communities/manager.go]: I was wondering about the scenario where two or more admin events come to control node before request to join:

1) acceptedRequestToJoin by adminA at timestamp 1
2) acceptedRequestToJoin by adminB at timestamp 2

at 2) `acceptedRequestsToJoin` will be populated with request that don't have revealed addresses. As a result, control node will reject the request before receiving addresses.

Comment (0x-r4bbit) [reply protocol/communities/manager.go]: So what I'd expect to happen in this case (and I'll add a test for this to verify this):

1. adminA decision reaches control node. Control node saves this request to join as it has seen it for the first time (at this point the request is in state acceptedPending
2. adminB's decision reaches control node. Control node sees that it already have a request with the same ID **and** it's in some pending state (acceptedPending or rejectedPending), so it'll ignore it altogether
3. Ultimately, control node will receive requestToJoin from user and perform the necessary checks. If there is already a request to join from an admin (meaning, the request exists in the database and is in any pending state coming from an admin), then it knows it can automatically accept (or reject) the request based on that

The important part here is that in 3) the request comes in via normal `HandleRequestToJoin()` while 1) and 2) are handled via `HandleCommunityEventsMessage`.

Comment (osmaczko) [reply protocol/communities/manager.go]: > 2. adminB's decision reaches control node. Control node sees that it already have a request with the same ID and it's in some pending state (acceptedPending or rejectedPending), so it'll ignore it altogether

I can't find this part in the code, could you please point me where this check happens?

Comment (0x-r4bbit) [reply protocol/communities/manager.go]: Ah, I'm relying on this https://github.com/status-im/status-go/pull/3843/files#diff-0c22647108517d38489ae5faa65a914ade3be02ea7a7e37355a82739da9aeb69R1411 which doesn't check for pending state. It'll basically ignore any additional incoming decision from an admin, regardless of the state of existing request. Should probably add a check for pending states there

Comment (osmaczko) [reply protocol/communities/manager.go]: > It'll basically ignore any additional incoming decision from an admin, regardless of the state of existing request

On the contrary, it will update the existing request if the new one (with higher clock value) comes :thinking: 

https://github.com/status-im/status-go/blob/bf60cd6a265495c677de9f0fce06e238670672d7/protocol/communities/manager.go#L1411-L1421

Comment (0x-r4bbit) [reply protocol/communities/manager.go]: > On the contrary, it will update the existing request if the new one (with higher clock value) comes 🤔

That's correct, but we expect it to be the same requestToJoin coming from different admins, so the clock will be the same no?

If you cancel your reques to join and send a new one, then it's in fact a new one which is fine (unless I'm missing something

Comment (osmaczko) [reply protocol/communities/manager.go]: > so the clock will be the same no?

Oh, I missed that. I assumed the clock is altered once admin accepts/rejects the request. All fine then, thanks for patience.

> The first one wins. But this can indeed cause a state where 2 admins look at different pending states until the ultimate state was confirmed by the control node.

That's acceptable to me.

**Review by osmaczko — APPROVED** [review]

File: protocol/communities/manager.go (hunk 12 above)
Comment (0x-r4bbit) [reply protocol/communities/manager.go]: Thank you for exploring this in detail! I'm going to add a test case for this regardless :)

File: protocol/communities/manager.go (hunk 13 above)
Comment (0x-r4bbit) [reply protocol/communities/manager.go:1395]: As discussed in https://github.com/status-im/status-go/pull/3843/files#r1282298019 we expect it to be the same clock, so we don't want to ignore this

File: protocol/communities/community.go (hunk 2 above)
Comment (0x-r4bbit) [reply protocol/communities/community.go]: Undoing this as I'm not handling kicking/banning members in this PR.

File: protocol/communities/manager.go (hunk 10 above)
Comment (0x-r4bbit) [reply protocol/communities/manager.go:1464]: After a few discussions with @osmaczko , this signal should be part of https://github.com/status-im/status-desktop/issues/11715, so that has to land first.

File: protocol/activity_center.go (hunk 1 above)
Comment (igor-sirotin) [reply protocol/activity_center.go:41]: Should it be `...AcceptPending` and `..DeclinePending` (without `ed`)?
Or am I misunderstanding smth?

Comment (0x-r4bbit) [reply protocol/activity_center.go:41]: Well the state is in "accepted pending" meaning, an admin has accepted the request but the decision is still pending until a control node confirms. Not sure if `-ed` is an issue here

File: protocol/communities_events_owner_without_community_key_test.go (hunk 11 above)
Comment (0x-r4bbit) [reply protocol/communities_events_owner_without_community_key_test.go]: I've updated the PR to make `additionalEventSenders` parameterizable so there's no need for `GetEventSender2()`, however, `setUpOnRequestCommunity()` now needs a `[]*Messenger{}` list of additional event senders where needed.

File: protocol/communities_events_utils_test.go [hunk 16]
```diff
@@ -363,9 +364,20 @@ func setUpOnRequestCommunityAndRoles(base CommunityEventsTestsInterface, role pr
...
+	for _, eventSender := range additionalEventSenders {
+		advertiseCommunityTo(s, community, base.GetControlNode(), eventSender)
+		joinOnRequestCommunity(s, community, base.GetControlNode(), eventSender)
+
+		grantPermission(s, community, base.GetControlNode(), eventSender, role)
+		checkPermissionGranted = func(response *MessengerResponse) error {
+			return checkRolePermissionInResponse(response, eventSender.IdentityPublicKey(), role)
+		}
+		waitOnMessengerResponse(s, WaitCommunityCondition, checkPermissionGranted, base.GetMember())
+		time.Sleep(500 * time.Millisecond)
```
Comment (mprakhov) [inline protocol/communities_events_utils_test.go]: this should work without sleep

Comment (0x-r4bbit) [reply protocol/communities_events_utils_test.go]: Ah, this has been flaky so I've added the sleep here, but didn't mean to leave it in.
I'm still making changes to this PR.

Any other ideas what can be done to make this less flaky?

Comment (mprakhov) [reply protocol/communities_events_utils_test.go]: 0. Need to detect where it is failing exactly
1. WaitCommunityCondition change on condition, where you will wait for a community + specific permission for a specific user
2. Call waitOnMessengerResponse for all other members in order for they have the latest CommunityDescription

P.S. If there will be any problems, ping me, I'll take a look at logs

File: protocol/communities/manager.go [hunk 17]
```diff
@@ -1812,22 +1958,42 @@ func (m *Manager) HandleCommunityRequestToJoin(signer *ecdsa.PublicKey, request
...
 		return requestToJoin, nil
 	}
 
-	permissionsSatisfied, _, err := m.accountsSatisfyPermissionsToJoin(community, request.RevealedAccounts)
-	if err != nil {
-		return nil, err
-	}
-	if !permissionsSatisfied {
-		requestToJoin.State = RequestToJoinStateDeclined
+	if community.IsControlNode() && len(request.RevealedAccounts) > 0 {
```
Comment (jrainville) [inline protocol/communities/manager.go:1963]: if `len(request.RevealedAccounts) == 0` and the community has permissions to join, we should auto set it to `RequestToJoinStateDeclined`. I'm not sure if the code does this.

File: protocol/messenger_communities.go [hunk 18]
```diff
@@ -284,6 +284,33 @@ func (m *Messenger) handleCommunitiesSubscription(c chan *communities.Subscripti
...
+				if sub.AcceptedRequestsToJoin != nil {
+					for _, requestID := range sub.AcceptedRequestsToJoin {
+						accept := &requests.AcceptRequestToJoinCommunity{
+							ID: requestID,
+						}
+						_, err := m.AcceptRequestToJoinCommunity(accept)
+						if err != nil {
+							m.logger.Warn("failed to accept request to join ", zap.Error(err))
+						}
+						// TODO INFORM ADMINS
```
Comment (jrainville) [inline protocol/messenger_communities.go:266]: Are those TODOs done or to be done in another PR?

**Review by jrainville — COMMENTED** [review]
Very good work. I had 2 questions.

File: protocol/messenger_communities.go (hunk 18 above)
Comment (0x-r4bbit) [reply protocol/messenger_communities.go:266]: This needs to be done as part of https://github.com/status-im/status-desktop/issues/11715

File: protocol/communities/manager.go (hunk 7 above)
Comment (0x-r4bbit) [reply protocol/communities/manager.go:1730]: Happening in https://github.com/status-im/status-go/pull/3853

File: protocol/communities/manager.go (hunk 17 above)
Comment (0x-r4bbit) [reply protocol/communities/manager.go:1963]: Ah actually, the check on `len(request.RevealedAccounts)` is a left over but it probably wouldn't hurt. It ends up being marked as `Declined` a few line below latest after the satisfaction check was done

**Review by MishkaRogachev — APPROVED** [review]
Great work! And no questions from my side :)

**Review by mprakhov — APPROVED** [review]

**Comment by alexjba** [comment]
There is one UI requirement that might need backed support and I can't spot it in this PR:

[Only the admin that made the initial decision can change the decision. (Other admins will not see the alternate button on hover).](https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba⎜Desktop?type=design&node-id=3113-356327&mode=design&t=Jij9C3tclFyDpj6t-0). Since only the initiator can see the alternate button and change the membership request state after accepting/rejecting it, we need to identify the source of membership state change. 

Was it considered for this PR?

**Comment by 0x-r4bbit** [comment]
@alexjba this is indeed not covered in this PR. The PR assumes that, once a state has been introduced by an admin, it can no longer be changed. 

@jrainville this would need change in the database and a migration script. Should this happen in this PR as well or should we create a separate issue for it so this can be tackled in follow-up work?

**Comment by jrainville** [comment]
> @jrainville this would need change in the database and a migration script. Should this happen in this PR as well or should we create a separate issue for it so this can be tackled in follow-up work?

It can be done in another ticket

**Comment by 0x-r4bbit** [comment]
> It can be done in another ticket

@jrainville added an issue describing what needs to be done here: https://github.com/status-im/status-desktop/issues/11842

Might be a good task for Teodor to get his feet wet with status-go
